### ISSUE TITLE
Freescout Inaccessible Elements For Screenreaders

### ISSUE THREAD

**Issue by fluxQuick840** [freescout-help-desk/freescout#4412]
After installing Freescout, I'm noticing a few inaccessibility issues in the app, and wanted to create an issue for the ones that I've found, as well as any others that may come to light in the future. So, just going in order here of where I noticed these:
1. On the Freescout settings page, there are two toggles, under the "Emails to Customers" heading, which I'm assuming are supposed to be checkboxes of some sort, which are not reported as such to my screenreader. They are instead read as blank lines, and are the elements right after the lines of text which say: "Open Tracking" and "Spread the Word".
2. The notifications submenu in the Freescout navigation menu is unlabeled for screenreaders. It is the only expandable menu which is unlabeled. The rest of them are fine.
3. On the "edit mailboxes" page, the editor allowing a formatted signature is not fully accessible. Clicking the "insert variable" button, for example, renders nothing that I've been able to interact with anywhere on the page. So, I can't insert any variables into my mailbox signature, which I'd like to be able to do.
4. After purchasing the "Time Tracking" module, all time tracking controls which are used to start, pause, and reset the timer in the conversation view are completely unlabeled. I have no idea which button is which, although I've sort of figured it out with trial and error. Still though, can we add titles to those or something?
I'm using the NVDA screenreader, which is free and open source if you'd like to try reproducing my findings.

**Comment by freescout-help** [comment]
FYI. We've pinged people in the previous thread https://github.com/freescout-help-desk/freescout/issues/1150 - maybe someone will be able to help.

1. We are not sure how to make such checkboxes visible. They do have "for" and "id" attributes. Maybe something is still missing.

![1](https://github.com/user-attachments/assets/5fdd7b5f-3032-4e5f-82e0-294ac58660ba)

2. `aria-label` attribute added to the notifications submenu.

3. Here we are also not sure how to make it visible. The "Insert variable" button here contains the `<select>` containing `<optgroup>` and `<options>` element.

![2](https://github.com/user-attachments/assets/fff46c3d-c338-441b-ae30-a6bc391bcf8a)

4. It will be fixed in the module.

**Comment by fluxQuick840** [comment]
Ahh, thank you for fixing the module and adding the ARIA label on that submenu!
I've done some fiddling around and I figured your checkbox issue out. All of these checkboxes are invisible to screenreaders because of this styling:
 .onoffswitch-checkbox {
    display: none;
}
You're essentially hiding it from the DOM and  it's not focusable. Maybe that  could be styled slightly differently to look the way you'd like while not setting the display to 'none'? This might possibly be your issue on the editor too, not sure as I haven't investigated that. 
Thanks again!

**Comment by freescout-help** [comment]
If you are able to edit the code of `/resources/views/settings/general.blade.php` can you try to add `aria-hidden="false"` to the `<input>` on line 125:
https://github.com/freescout-help-desk/freescout/blob/dist/resources/views/settings/general.blade.php#L125

**Comment by fluxQuick840** [comment]
Yes, I can do that. It didn't help unfortunately, though. There is still no focusable checkbox there in settings.

**Comment by freescout-help** [comment]
Try to change the code in `/public/css/style.css` on line 3178 to the following:

```
.onoffswitch-checkbox {
    position: absolute;
    left:-9999px;
}
```

**Comment by fluxQuick840** [comment]
Yes, that does it. Now the display isn't set to none, my screenreader can focus on the checkboxes. I can interact with them without issue. Thanks!
Let's try to work on the "insert variable" selector menu in the editor now. I notice that the tab index on that select element is set to "-1", which I believe means that it isn't part of the tab order and can't be focused on using the keyboard. I'm not a hundred percent on that, but I wonder if it might be contributing to my issue finding it after I use the button to bring it up. Thoughts?

**Comment by freescout-help** [comment]
Try to remove `tabindex="-1"` on line 200 in `/public/js/main.js` and [Clear Cache](https://github.com/freescout-help-desk/freescout/wiki/Clearing-the-Cache).

**Comment by fluxQuick840** [comment]
Boom! It was that easy! Now, after clicking the button, I can press tab and interact with the dropdown without issue! Thanks so much! 
As of now, this resolves the accessibility issues I was having, so long as I'll be able to update my time tracking module soon and have those buttons labeled. Does GitHub remain the best place to report accessibility issues for modules, by the way?
Thank you again for all your help!

**Comment by freescout-help** [comment]
Yes, you can keep reporting here.
